app/code/Magento/Catalog/Test/Unit/Controller/Category/ViewTest.php

```diff
@@ -199,11 +205,45 @@ public function testApplyCustomLayoutUpdate()
             \Magento\Framework\DataObject::class,
             ['getPageLayout', 'getLayoutUpdates']
         );
+        $this->expectationForPageLayoutHandles($expectedData);
         $settings->expects($this->atLeastOnce())->method('getPageLayout')->will($this->returnValue($pageLayout));
         $settings->expects($this->once())->method('getLayoutUpdates')->willReturn(['update1', 'update2']);
-
         $this->catalogDesign->expects($this->any())->method('getDesignSettings')->will($this->returnValue($settings));
 
         $this->action->execute();
     }
+
+    /**
+     * Expected invocation for Layout Handles
+     *
+     * @param array $data
+     * @return void
+     */
+    private function expectationForPageLayoutHandles($data): void
+    {
+        $index = 2;
+        foreach ($data as $expectedData) {
+            $this->page->expects($this->at($index))
+            ->method('addPageLayoutHandles')
+            ->with($expectedData[0], $expectedData[1], $expectedData[2]);
+            $index++;
+        }
+    }
+
+    /**
+     * Data provider for execute method.
+     *
+     * @return array
+     */
+    public function getInvocationData(): array
+    {
+        return [
```

Comment:
hi, would you consider expanding the test to check different page types and display modes?

Reply:
Yes sure :+1: